### ISSUE TITLE
convert-pretrain-model-to-d2.py working poorly with Panoptic-DeepLab X65-DC5 model

### ISSUE THREAD

**Issue by mike239x** [bowenc0221/panoptic-deeplab#56]
When working with Panoptic-DeepLab X65-DC5 model the following happens:

in [convert-pretrain-model-to-d2.py line 27](https://github.com/bowenc0221/panoptic-deeplab/blob/master/tools_d2/convert-pretrain-model-to-d2.py#L27) `torch.load` returns a dict with keys `'model', 'optimizer', 'scheduler', 'iteration'`, and then on line 29 that is put into another dict with keys `'model', '__author__', 'matching_heuristics'`. After it is loaded into the [`DefaultPredictor`](https://github.com/facebookresearch/detectron2/blob/4aa1af8c5d8f6182b2cacd82d8ead153d7a5f9e7/detectron2/engine/defaults.py#L160) from detectron2 that would try to construct [`DetectionCheckpointer`](https://github.com/facebookresearch/detectron2/blob/4aa1af8c5d8f6182b2cacd82d8ead153d7a5f9e7/detectron2/checkpoint/detection_checkpoint.py#L11) which would run the lines
```python3
        checkpoint = self._load_file(path)
        incompatible = self._load_model(checkpoint)
```
from the [base class](https://github.com/facebookresearch/fvcore/blob/master/fvcore/common/checkpoint.py#L123).
`_load_file` just unpickles that dict that we created before, and  `_load_model` runs
`self._convert_ndarray_to_tensor(checkpoint["model"])`, which is gonna fail since `checkpoint["model"]` is a dict with another `'model'` key in it.

Long story short either the model is not stored correctly or the conversion script did not flatten the dicts properly - I blame the latter.

To reproduce:
get the [X65-DC5 model](https://drive.google.com/file/d/1F9Biuu9UmgfCyatP5yQTYN5V5_YyVGA2/view);
in `panoptic-deeplab/tools_d2` run `python3 convert-pretrain-model-to-d2.py <model.pth> x65.pkl`;
run something like this in python:
```python3
from detectron2.checkpoint.detection_checkpoint import DetectionCheckpointer
d = DetectionCheckpointer('x65.pkl')
```

**Comment by bowenc0221** [comment]
Check: https://github.com/facebookresearch/detectron2/blob/master/tools/convert-torchvision-to-d2.py

**Comment by mike239x** [comment]
Well, I don't think that link you sent me is very helpful, since there the same problem remains - `data['model']` is not an ndarray. As for myself, I resolved my problem by changing line 29 of convert-pretrain-model-to-d2.py from
```python3
    res = {"model": obj, "__author__": "third_party", "matching_heuristics": True}
```
to
```python3
    res = {**obj, "__author__": "third_party", "matching_heuristics": True}
```
and that seems to solve my problems just fine.
